README.md

```diff
@@ -25,74 +30,78 @@ reactivebleclient.scanForDevices(withService: uuid, scanMode: ScanMode.lowLatenc
 
 There are numerous issues on the Android BLE stack that leave it hanging when you try to connect to a device that is not in range. To work around this
-issue use the method `connectToAdvertisingDevice` to first scan for the device and when it is found connect to it.
+issue use the method `connectToAdvertisingDevice` to first scan for the device and only if it is found connect to it.
 
 ```dart
-  reactiveBleClient.connectToAdvertisingDevice(
-      id: 'AA:BB:CC:DD:EE:FF', 
-      withService: serviceUuid,
-      prescanDuration: const Duration(seconds: 5), 
-      servicesWithCharacteristicsToDiscover: {serviceUuid: [char1, char2]},
-      connectionTimeout: const Duration(seconds:  2),
-    ).listen((connectionUpdate) {
-      //code to handle connection update
-    }, onError: (dynamic error) {
-      //code to handle error
-    });
+reactiveBleClient.connectToAdvertisingDevice(
+    id: foundDeviceId,
+    withService: serviceUuid,
+    prescanDuration: const Duration(seconds: 5),
+    servicesWithCharacteristicsToDiscover: {serviceId: [char1, char2]},
+    connectionTimeout: const Duration(seconds:  2),
+  ).listen((connectionState) {
+    // Handle connection state updates
+  }, onError: (dynamic error) {
+    // Handle a possible error
+  });
 ```
-Besides the normal connection parameters that are described at above this function also has 2 additional required parameters: `withService` and  `prescanDuration`. PreScanDuration is the amount of time the ble stack will scan for the device before it attempts to connect (if the device is found)
+
+Besides the normal connection parameters that are described above this function also has 2 additional required parameters: `withService` and  `prescanDuration`. PreScanDuration is the amount of time the ble stack will scan for the device before it attempts to connect (if the device is found)
 
 ### Read / write characteristics
 
 #### Read characteristic
+
 ```dart
-final characteristic = QualifiedCharacteristic(serviceId: serviceUuid, characteristicId: characteristicUuid, deviceId: 'AA:BB:CC:DD:EE:FF');
+final characteristic = QualifiedCharacteristic(serviceId: serviceUuid, characteristicId: characteristicUuid, deviceId: foundDeviceId);
 final response = await reactiveBleClient.readCharacteristic(characteristic);
 ```
 
 #### Write with response
-Write a value to characteristic and await the reaponse. The return value is a `List<Int>`.
+
+Write a value to characteristic and await the response. The return value is a list of bytes (`List<int>`).
```

Comment:
Actually, I think we should switch to `Uint8List`, because it suggests it means exactly "list of bytes", whereas `List<int>` is ambiguous. `Uint8List` is also a `TypedData`, which is useful when decoding blobs.